docs/nf_templates/8_components.md

```diff
@@ -0,0 +1,167 @@
+
+```
+nf-core modules install
+```
+
+You can follow the prompts to find and install the module you are interested in:
+
+```console
+? Tool name: fastp
+```
+
+Once selected, the tooling will install the module in `modules/nf-core/` folder and give you a suggested line of code for your main workflow file (e.g., `workflows/mypipeline.nf`).
+
+```console
+INFO     Installing 'fastp'
+INFO     Use the following statement to include this module:
+
+include { FASTP } from '../modules/nf-core/fastp/main'
+```
+
+!!! question "Exercise"
+
+    Install the `fastp` module to your pipeline.
+
+!!! note "What is fastp?
+
+    Fastp is a...
+
+### Adding a module to your pipeline
+
+Although the modules has been installed in your repository, it is not yet added to your pipeline. The suggested `include` statement needs to be added to your `workflows/mypipeline.nf` file.
+
+```console
+include { FASTP } from '../modules/nf-core/fastp/main'
+```
+
+!!! question "Exercise"
+
+    Add the `include { FASTP } from '../modules/nf-core/fastp/main'` statement to your `workflows/mypipeline.nf` file.
+
+To implement the module in your pipeline you will need to check what inputs are required. You can view the inputs channels for the module by opening the `/modules/nf-core/fastp/main` file.
+
+```groovy title="/modules/nf-core/fastp/main.nf"
+input:
+tuple val(meta), path(reads)
+path  adapter_fasta
+val   save_trimmed_fail
+val   save_merged
+```
+
+Each module has a meta file which describes the required inputs. The meta file is rendered on the [nf-core website](https://nf-co.re/modules/fastp), or can be viewed using the `nf-core modules info command`.
+
+Code for `workflows/mypipeline.nf`
+
+!!! question "Exercise"
+
+    Add the following code to your `workflows/mypipeline.nf` file to include the fastp process in your pipeline.
+
+```groovy title="workflows/mypipeline.nf"
+//
+// MODULE: Run Fastp
+//
+FASTP (
+    ch_samplesheet,
+    [],
+    false,
+    false
+)
+```
+
+!!! note "What is `[]`?"
+
+    Every channel needs to be filled for a process to run. In the example above `[]` was used to fill the channel with an empty list.
+
+Importantly, nf-core components are tracked in `modules.json`. Tracking components improves reproducibility and reporting by giving some certainty about the version of the tool you were using. The `modules.json` will be updated with a hash for the module.
+
+!!! tip "Private modules repositories"
+
+    Private repositories of modules can be accessed using `--git-remote` in `nf-core modules install` command.
+
+### Linting modules
+
+The `nf-core lint` command will help manage nf-core components and test that they match the remote they came from. Although modules are written to be flexible, you may want to modify a component to fit your purpose.
+
+For example, if you modify an nf-core module, it will no longer match the remote and a linting test of this module would fail.
+
+```
+EXAMPLE
```

Comment:
Reminder to fill this out with the example.

Reply:
Thanks!